### DISCUSSION TITLE
Allow Checkmark Widget to show current streak instead of current score

### DISCUSSION THREAD

**Discussion by Teaminds** [iSoron/uhabits#342] [Feature Requests]
I was very motivated by the idea of "unbroken chain" - that is, the knowledge that I was "doing" the problem without permits has been so many times in a row. For example, "You publish records on Facebook for 24 days" - it literally controls me.
I'm happy to use the application on my Android smartphone, but I'm very uncomfortable that on the screen with widgets-habits (I have a lot of them) I do not see the chain length for every habit (and, ideally, how many days before the break is left - I there are a few habits "once a week" and "4 times in 30 days").

**Comment by iSoron** [comment]
Thank you for the suggestion, @Teaminds. I guess we could allow the user to choose whether the checkmark widget should display the current score or the current streak. In case we show the current streak, it would probably also be a good idea to show the best streak so far. I am not sure how to represent this visually, though. I would like to avoid having simply a pair of numbers, if possible. If anyone has any ideas and want to create a quick mock-up, that would be very welcome.

**Comment by mrgoodrich** [comment]
+1 to this suggestion. Regarding the visualization, as a user I would be fine with numbers for the minimum viable product @iSoron , especially if it was an option in settings. The numbers could appear after the checkmark is clicked.

**Comment by dkabambe** [comment]
+1 from me too. I'd be happy to just have current steak on widget as that's the key at a glance info.

Would be good to have current and longest steak on the main home screen, (maybe only in landscape, on bigger displays, or as optional second line for each habit?)

**Comment by spitofland** [comment]
I think just the current streak would be the most relevant information to the checkbox widget, because it's what you are currently working on. Trying to fit the best streak as well might get tight. Maybe have an option to just include the current streak and another option to also include the best streak.

For display, I think the best option would be to put the number(s) in the middle of the circle where the 'x' and check mark currently sit. If both numbers are displayed, I would present it like a fraction (current/best with a horizontal line between them). To help users understand, you really only need to use a similar notation inside the app. For example, add the fraction (current/best) on the main screen after the habit name, to the "Overview" on the habit page, and/or in the "Best streaks" section of the habit page. You could also tell the user what it will look like in the settings with something like "Show the Current and Best Streaks in the checkbox widget as a fraction (current over best)".

I would always show these numbers for motivation. The color coding change should be enough to signal that the box has been checked, though you will want to check with some color-blind users.

**Comment by NegatioN** [comment]
A lot of you mention a need for Current Streak shown in contrast to the Best Streak. This seems a bit like scope creep on a somewhat simple feature, and I (personally) don't really see the need for this, especially considering there's such limited real-estate in the 1x1 square.

I think simply showing the current streak looks great, no?
![number2](https://user-images.githubusercontent.com/4037769/84966375-adffbc00-b111-11ea-8650-ab45d8c08537.png)


If the plan is to allow user-customization here, this would still be viable as a first implementation?

I'd also like to mention my suggestion in #597 as a solution to "being unable to see how many days are left before the chain breaks"

EDIT: I see now that I actually misunderstood what this request was about. :slightly_smiling_face: I would still like to pitch the idea of having the option to display the current streak instead of a checkmark though
